packages/block-editor/src/components/global-styles/use-global-styles-output.js

```diff
@@ -830,39 +836,34 @@ export const toStyles = (
 							// Otherwise add regular selectors.
 							const styleVariationDeclarations =
 								getStylesDeclarations(
-									styles?.variations?.[ styleVariationName ],
+									styleVariations,
 									styleVariationSelector,
 									useRootPaddingAlign,
 									tree
 								);
-							if ( !! styleVariationDeclarations.length ) {
-								ruleset =
-									ruleset +
-									`${ styleVariationSelector }{${ styleVariationDeclarations.join(
-										';'
-									) }}`;
+							if ( styleVariationDeclarations.length ) {
+								ruleset += `${ styleVariationSelector }{${ styleVariationDeclarations.join(
+									';'
+								) };}`;
 							}
 						}
 					}
 				);
 			}
 
-			const duotoneStyles = {};
-			if ( styles?.filter ) {
-				duotoneStyles.filter = styles.filter;
-				delete styles.filter;
-			}
-
```

Comment:
@ajlende This behavior, where the duotone style is always extracted, and then used only conditionally, when the duotone selector exists, I think it was a bug originally introduced in #37727. It means that if there is a `filter.duotone` style specified, and no feature selector for it, then the style will be ignored. Instead of being applied to the root selector just like any other style. Not sure if this was ever intended.

The duotone selector was a first ever example of a "feature selector". And the generic feature selector feature was introduced only later. Today, it might be possible to remove the duotone-specific code completely and just use a generic feature selector for it. The only thing that needs to be done is to convert the legacy `__experimentalDuotone` option into a featured `selectors.filter.duotone` selector.

Reply:
Should we go ahead and actually do that?

Reply:
By "that" you mean eliminating the `duotoneSelector` variable and integrate it into `featureSelectors`? Yes, I could push a commit like this :+1:

Reply:
yes, that's the idea. 

Potentially we could add a "shim" within `registerBlockType` + a deprecated function call.